lib/cache/storages/MemoryStorage.ts

```diff
@@ -0,0 +1,50 @@
+import { Storage } from './Storage';
+
+export class MemoryStorage implements Storage {
+
+  private readonly storage = new Map<string, any>();
+  private keys: string[];
+  private readonly hasLimit: boolean;
+
+  constructor(private readonly limit?: number) {
+    this.hasLimit = typeof this.limit === 'number';
+    if (this.hasLimit) {
+      this.keys = [];
+    }
+  }
+
+  public set<V>(key: string, value: V): Promise<void> {
+    this.checkSize(key);
+
+    this.storage.set(key, value);
+
+    return Promise.resolve();
+  }
+
+  public get<V>(key: string): Promise<V> {
+    return Promise.resolve(this.storage.get(key));
+  }
+
+  public delete(key: string = this.keys[0]): Promise<void> {
```

Comment:
Using a default value for `key` here is not intuitive. It is recommended to keep the same signature as in the original interface.